module3/homework/advanced_calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,61 @@
+#include "advancedCalculator.hpp"
+#include "calcExceptions.hpp"
+
+#include <iostream>
+#include <map>
+#include <functional>
+#include <math.h>
+
+FunctionMap functions{
+        {'+', std::plus<double>()},
+        {'-', std::minus<double>()},
+        {'*', std::multiplies<double>()},
+        {'/', division},
+        {'!', factorial},
+        {'^', power},
+        {'$', squareRoot},
+        {'%', modulo}
+};
+
+double division(double a, double b){
+    if(b == 0)
+        throw DivideBy0Exception();
+    return a / b;
+}
+
+double factorial(double a, double b){
+    if(a < 0)
+        return -tgamma(-a + 1);
+    return tgamma(a + 1);
```

Comment:
```suggestion
    return a < 0 ? -tgamma(-a + 1) :  tgamma(a + 1);
```
This is nice place to use lambas.

Reply:
This is not lambda, it's just short if statement. But good place to refactor :)